src/explore-education-statistics-admin/src/pages/release/content/components/ReleaseContentAccordionSection.tsx

```diff
@@ -123,100 +144,127 @@ const ReleaseContentAccordionSection = ({
+                  visible={open}
                 />
               )}
+              renderEditableBlock={block => (
+                <ReleaseEditableBlock
+                  allowComments
+                  allowImages
+                  block={block}
+                  sectionId={sectionId}
+                  sectionKey="content"
+                  editable={!isReordering}
+                  publicationId={release.publication.id}
+                  releaseId={release.id}
+                  visible={open}
+                />
+              )}
+            />
 
-              <ButtonGroup className="govuk-!-margin-bottom-8 dfe-justify-content--center">
-                <Button variant="secondary" onClick={addBlock}>
-                  Add text block
-                </Button>
-                {!showDataBlockForm && (
-                  <Button variant="secondary" onClick={toggleDataBlockForm.on}>
-                    Add data block
-                  </Button>
+            {editingMode === 'edit' && !isReordering && (
+              <>
+                {showDataBlockForm && (
+                  <DataBlockSelectForm
+                    id={`dataBlockSelectForm-${sectionId}`}
+                    releaseId={release.id}
+                    onSelect={async selectedDataBlockId => {
+                      await attachDataBlock(selectedDataBlockId);
+                      toggleDataBlockForm.off();
+                    }}
+                    onCancel={toggleDataBlockForm.off}
+                  />
                 )}
-              </ButtonGroup>
-            </>
-          )}
-        </>
-      )}
-    </EditableAccordionSection>
+
+                <ButtonGroup className="govuk-!-margin-bottom-8 dfe-justify-content--center">
+                  <Button variant="secondary" onClick={addBlock}>
+                    Add text block
+                  </Button>
+                  {!showDataBlockForm && (
+                    <Button
+                      variant="secondary"
+                      onClick={toggleDataBlockForm.on}
+                    >
+                      Add data block
+                    </Button>
+                  )}
+                  {!showEmbedDashboardForm && (
+                    <Button
+                      variant="secondary"
+                      onClick={toggleEmbedDashboardForm.on}
+                    >
+                      Embed dashboard
+                    </Button>
+                  )}
+                </ButtonGroup>
+              </>
+            )}
+          </>
+        )}
+      </EditableAccordionSection>
+
+      <Modal
+        title="Embed dashboard link"
+        open={showEmbedDashboardForm}
+        onExit={toggleEmbedDashboardForm.off}
+      >
+        <EditableEmbedForm
+          onCancel={toggleEmbedDashboardForm.off}
+          onSubmit={embedBlock => {
+            addEmbedBlock(embedBlock);
+            toggleEmbedDashboardForm.off();
+          }}
```

Comment:
Similar to another comment, would suggest we async/await the `addEmbedBlock` call to avoid the toggle being flipped before the work has actually finished